### PR TITLE
fix: submit admin feedback form

### PR DESCRIPTION
**Pull request by wanlingt**
## Problem

The feedback form for admins (on admin pages) was not being submitted as there were some form fields that did not exist on the admin feedback form, e.g. `attachmentType`. These fields were returning `undefined` and prevented the form from being submitted successfully.

## Solution

Vary the form fields depending on whether the feedback form is an admin's or public user's.

**Breaking Changes** 

- [x] No - this PR is backwards compatible  

**AFTER**:

chrome-capture-2023-2-23

## Tests

- [x] On an admin page, click the speech bubble icon on the bottom right of the screen. Submit your feedback (make sure that any text entered is longer than 5 characters)
- [x] (To test that the public feedback form is also still working) Change `REACT_MIGRATION_RESP_ROLLOUT_EMAIL` to 99. Go to an email-mode public form and click the "switch back" banner at the bottom of the page. This should open a modal with long text field (and email field if not logged in as admin). Fill the form. It should submit successfully to the public feedback form.

### CODE REVIEW

**Review by justynoh — APPROVED** [review]
lgtm

File: frontend/src/features/env/AdminFeedbackModal.tsx [hunk 1]
```diff
@@ -68,7 +68,9 @@ export const AdminFeedbackModal = ({
   const feedbackMutation = useFeedbackMutation()
 
   const handleSubmitForm = handleSubmit((formInputs: AdminFeedbackFormDto) => {
-    if (!feedbackForm) return
+    if (!feedbackForm) {
```
Comment (foochifa) [inline frontend/src/features/env/AdminFeedbackModal.tsx]: nit: but is this change required?

File: frontend/src/features/env/EnvService.ts [hunk 2]
```diff
@@ -88,17 +88,10 @@ export const submitSwitchEnvFormFeedback = async ({
...
-    'rumSessionId',
-    'attachmentType',
-    'userAgent',
-    'responseMode',
-    'authType',
-  ]
+  const formFields = ['url', 'feedback', 'email', 'rumSessionId']
   if (isAdmin) formFields.push('rating')
+  if (!isAdmin)
+    formFields.push('attachmentType', 'userAgent', 'responseMode', 'authType')
```
Comment (foochifa) [inline frontend/src/features/env/EnvService.ts]: Another nit, should we abstract this to make it more verbose?

like 
```
adminFormFields = ['rating']
publicFormFields = ['attachmentType', 'userAgent', 'responseMode', 'authType']
if (isAdmin) formFields.push(...adminFormFields)
else formFields.push(...publicFormFields)
```

**Review by foochifa — APPROVED** [review]
else lgtm

File: frontend/src/features/env/EnvService.ts (hunk 2 above)
Comment (wanlingt) [reply frontend/src/features/env/EnvService.ts]: [20bd598](https://github.com/opengovsg/FormSG/pull/5981/commits/20bd598415531c3246827887baed86276c104147)

File: frontend/src/features/env/AdminFeedbackModal.tsx (hunk 1 above)
Comment (wanlingt) [reply frontend/src/features/env/AdminFeedbackModal.tsx]: [20bd598](https://github.com/opengovsg/FormSG/pull/5981/commits/20bd598415531c3246827887baed86276c104147)
